CODE_GENERATION.md

```diff
@@ -0,0 +1,290 @@
+# Contributing New Resource Using ACK
```

Comment:
Nit: I bet 90% of folks reading this will know already, but it may be worth using 1-2 sentences to explain what ACK is and how it relates to Crossplane (e.g. that they share a codegen pipeline but the actual Amazon controllers are a separate thing).